code/game/machinery/bots/medbot.dm

```diff
@@ -507,7 +531,23 @@
 	visible_message("<span class='userdanger'>[src] blows apart!</span>")
 	var/turf/Tsec = get_turf(src)
 
-	new /obj/item/weapon/storage/firstaid(Tsec)
+	switch(skin)

```

Comment:
I did intend to come back and write something more than
>eww

You can store a typepath of the used medkit, and have medkits have a ```on_bot_state``` var and use ```initial()``` to grab it from the typepath.